src/test/scala/org/broadinstitute/dsde/firecloud/service/WorkspaceServiceSpec.scala

```diff
@@ -0,0 +1,80 @@
+package org.broadinstitute.dsde.firecloud.service
+
+import org.broadinstitute.dsde.firecloud.MockServers
+import org.broadinstitute.dsde.firecloud.model.ModelJsonProtocol._
+import org.broadinstitute.dsde.firecloud.model.{WorkspaceEntity, WorkspaceIngest}
+import org.broadinstitute.dsde.vault.common.openam.OpenAMSession
+import org.scalatest.concurrent.ScalaFutures
+import org.scalatest.time.{Seconds, Span}
+import org.scalatest.{FreeSpec, Matchers}
+import spray.http.HttpCookie
+import spray.http.HttpHeaders.Cookie
+import spray.http.StatusCodes._
+import spray.httpx.SprayJsonSupport._
+import spray.testkit.ScalatestRouteTest
+
+class WorkspaceServiceSpec extends FreeSpec with ScalaFutures with ScalatestRouteTest with Matchers with WorkspaceService {
+
+  def actorRefFactory = system
+
+  private final val ApiPrefix = "/workspaces"
+
+  override def beforeAll(): Unit = {
+    MockServers.startWorkspaceServer()
+  }
+
+  override def afterAll(): Unit = {
+    MockServers.stopWorkspaceServer()
+  }
+
+  "WorkspaceService" - {
+
+    val openAMSession = OpenAMSession(()).futureValue(timeout(Span(5, Seconds)), interval(scaled(Span(0.5, Seconds))))
+    val token = openAMSession.cookies.head.content
+    val workspaceIngest = WorkspaceIngest(
+      name = Some(MockServers.randomAlpha()),
+      namespace = Some(MockServers.randomAlpha()))
+
+    "when calling POST on the workspaces path with a valid WorkspaceIngest" - {
```

Comment:
I'd love to see an additional test that POSTs an invalid/incomplete WorkspaceIngest.